physicsnemo/models/dlwp_healpix_layers/healpix_blocks.py

```diff
@@ -596,103 +631,154 @@ def __init__(
+            if batch_norm else None
         )
-        if activation is not None:
-            convblock.append(activation)
-        # 1x1 convolution returning to latent channels
-        convblock.append(
-            geometry_layer(
-                layer=torch.nn.Conv2d,
-                in_channels=int(latent_channels * upscale_factor),
-                out_channels=int(latent_channels),
-                kernel_size=1,
-                dilation=dilation,
-                enable_nhwc=enable_nhwc,
-                enable_healpixpad=enable_healpixpad,
-            )
+        # Conditional Layer Normalization
+        if conditional_layer_norm is not None:
+            # resolve context-dependent parameters
+            self.cln = conditional_layer_norm
+            self.cln.setup_cln(channel_depth=int(latent_channels))
+        else:
+            self.cln = None
+        # self.cln = conditional_layer_norm if conditional_layer_norm is not None else None
+
+        self.dropout_layer = (
+            th.nn.Dropout2d(p=dropout)
+            if self.dropout else None
         )
-        if activation is not None:
-            convblock.append(activation)
-        # 3x3 convolution from latent channels to latent channels
-        convblock.append(
-            geometry_layer(
-                layer=torch.nn.Conv2d,
-                in_channels=int(latent_channels),
-                out_channels=out_channels,  # int(latent_channels),
-                kernel_size=kernel_size,
-                dilation=dilation,
-                enable_nhwc=enable_nhwc,
-                enable_healpixpad=enable_healpixpad,
-            )
+
+        # 1x1: latent → latent * upscale
+        self.expand_conv = geometry_layer(
+            layer=th.nn.Conv2d,
+            in_channels=int(latent_channels),
+            out_channels=int(latent_channels * upscale_factor),
+            kernel_size=1,
+            dilation=dilation,
+            enable_nhwc=enable_nhwc,
+            enable_healpixpad=enable_healpixpad,
         )
-        if activation is not None:
-            convblock.append(activation)
-        self.convblock = th.nn.Sequential(*convblock)
 
-    def forward(self, x):
-        """Forward pass of the SymmetricConvNextBlock
+        # 1x1: upscale → latent
+        self.reduce_conv = geometry_layer(
+            layer=th.nn.Conv2d,
+            in_channels=int(latent_channels * upscale_factor),
+            out_channels=int(latent_channels),
+            kernel_size=1,
+            dilation=dilation,
+            enable_nhwc=enable_nhwc,
+            enable_healpixpad=enable_healpixpad,
+        )
 
+        # 3x3: latent → out
+        self.output_conv = geometry_layer(
+            layer=th.nn.Conv2d,
+            in_channels=int(latent_channels),
+            out_channels=out_channels,
+            kernel_size=kernel_size,
+            dilation=dilation,
+            enable_nhwc=enable_nhwc,
+            enable_healpixpad=enable_healpixpad,
+        )
```

Comment:
I understand why you had to unroll what was previously a simple `nn.Sequential` into these separate modules, to support passing the conditional arg. It's a bit of a pain, but for backwards compatibility of checkpoints we could consider collecting things into a `nn.ModuleList` that is still named `self.convblock`. I think then old checkpoints would load without issue, and the core part of forward would look something like:
```
for layer in self.convblock:
    if isinstance(layer, ConditionalLayerNorm):
        x = layer(x, conditions_cln)
    else:
        x = layer(x)
```
Thoughts?